homeassistant/components/onkyo/media_player.py

```diff
@@ -162,116 +198,100 @@ def determine_zones(receiver):
     return out
 
 
-def setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up the Onkyo platform."""
+    """Set up the Onkyo Platform from config_flow."""
+
     hosts: list[OnkyoDevice] = []
+    platform = entity_platform.async_get_current_platform()
 
-    def service_handle(service: ServiceCall) -> None:
+    def select_output_service(
+        entity: OnkyoDevice, call: ServiceCall
+    ) -> ServiceResponse:
         """Handle for services."""
-        entity_ids = service.data[ATTR_ENTITY_ID]
-        devices = [d for d in hosts if d.entity_id in entity_ids]
-
-        for device in devices:
-            if service.service == SERVICE_SELECT_HDMI_OUTPUT:
-                device.select_output(service.data[ATTR_HDMI_OUTPUT])
+        entity.select_output(call.data[ATTR_HDMI_OUTPUT])
+        return None
 
-    hass.services.register(
-        DOMAIN,
-        SERVICE_SELECT_HDMI_OUTPUT,
-        service_handle,
+    platform.async_register_entity_service(
+        name=SERVICE_SELECT_HDMI_OUTPUT,
         schema=ONKYO_SELECT_OUTPUT_SCHEMA,
+        func=select_output_service,
     )
```

Comment:
I made it an entity service, since this is no longer a platform integration. I believe that's due to it being a device instead of a hub. I also renamed the callback name to be more specific.

Maybe I can convert it back to a hub <sub>that would pain my heart, but motivates me to create a follow up pretty much instantly :-p</sub>

Reply:
I think the former, hass.services.register would be more correct. I think the endgoal of this PR should be that the service doesn't change in the way its called

Reply:
Fair point. I'll see what I can do